### PR TITLE
SSR: Fixed broken reflections on OpenGL (close #151)

### LINKED ISSUE #151: SSR: Latest changes broke reflections on OpenGL
This commit broke reflections on OpenGL
https://github.com/DiligentGraphics/DiligentFX/commit/64deda8d453f7db4eb7db23c31759cbf960ac9fa

### PR DESCRIPTION
**Pull request by MikhailGorobets**

### CODE REVIEW

File: Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputeAmbientOcclusion.fx [hunk 1]
```diff
@@ -33,7 +33,7 @@ float3 SampleNormalWS(float2 ScreenCoordUV)
 
 float SamplePrefilteredDepth(float2 ScreenCoordUV, float MipLevel)
```
Comment (TheMostDiligent) [inline Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputeAmbientOcclusion.fx:34]: The function name is now misleading - it returns Z, not depth.
Please rename this function and others to reflect what they actually do, or alternatively, convert depth to z where the return value is used.

**Comment by MikhailGorobets** [comment]
> How do changes to SSAO shaders fix reflections?
Changed `FastReconstructPosition` function which is used in SSAO as well

File: Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputeAmbientOcclusion.fx (hunk 1 above)
Comment (MikhailGorobets) [reply Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputeAmbientOcclusion.fx:34]: @TheMostDiligent 
The function returns a hyperbolic depth in NDC space, not a linear depth. It is only necessary for OpenGL

**Review by TheMostDiligent — CHANGES_REQUESTED** [review]
This PR confuses Depth with NDC Z which makes reading the shader very difficult and is the source of errors.
It needs to be reworked to make sure variable names are consistent

- `Depth` is what is read from the Depth buffer. It is always in the range [0, 1]
- `NdcZ` may be in range [-1, +1] or [0, +1]
- It is not clear what `FastReconstructPosition` takes and what returns. It should be renamed to something `ScreenXYDepthToViewSpace`
- `SamplePrefilteredDepth` and other similar functions should be kept as is. Transformation from Depth to NDC should be done in `ScreenXYDepthToViewSpace`

**Pull request merged**